### PR TITLE
Added a section on Dependency Injection technology in documentation

### PR DESCRIPTION
**Pull request by wakaree**
# Description

I often noticed that developers ignored the existence of the concept of dependency injection, using global variables or singleton instead. Previously, this topic was not covered in the documentation, which I corrected.

## Type of change

- [x] Documentation (typos, code examples or any documentation update)

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/dispatcher/dependency_injection.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+######
```
Comment (JrooTJunior) [inline docs/dispatcher/dependency_injection.rst:1]: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

> Section headers ([ref](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#sections)) are created by underlining (and optionally overlining) the section title with a punctuation character, **at least as long as the text**

File: docs/dispatcher/dependency_injection.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+Dependency injection
```
Comment (JrooTJunior) [inline docs/dispatcher/dependency_injection.rst:2]: I don't think so this is a clear dependency injection mechanism, but i don't know another name for this feature

File: examples/context_addition_from_filter.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+class HelloFilter(Filter):
+    def __init__(self, name: Optional[str] = None) -> None:
+        self.name = name
+
+    async def __call__(
+        self,
+        message: Message,
+        event_from_user: User
+        # as I said previously, filters also can accept keyword parameters like in handlers
```
Comment (JrooTJunior) [inline examples/context_addition_from_filter.py:19]: "As I said previously" is not clear sentence for this place in due to example can be viewed without reading documentation and this comment is first in this file, you say nothing previously.

File: examples/context_addition_from_filter.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    ) -> Union[bool, Dict[str, Any]]:
+        if message.text.casefold() == "hello":
+            # returning a dictionary that will update the context data
+            return {"name": event_from_user.mention_html(name=self.name)}
+        return False
+
+
+@router.message(HelloFilter(name=None))
```
Comment (JrooTJunior) [inline examples/context_addition_from_filter.py:27]: name is optional argument, you shouldn't pass any balue to it

**Review by JrooTJunior — CHANGES_REQUESTED** [review]

File: docs/dispatcher/dependency_injection.rst (hunk 1 above)
Comment (wakaree) [reply docs/dispatcher/dependency_injection.rst]: Fixed

File: docs/dispatcher/dependency_injection.rst (hunk 2 above)
Comment (wakaree) [reply docs/dispatcher/dependency_injection.rst:2]: > I don't think so this is a clear dependency injection mechanism
The basic concept of dependency injection is clearly visible in the example of passing a router to the current handler (by "event_router" keyword) - the handler itself depends on it and has access to it. 

But you really can pass anything through the handling context and do it both at application startup and at runtime. Nevertheless, I saw other libraries that also name the concept of passing arguments to the function by parameter names / type hints like "depends".

> but i don't know another name for this feature
Maybe "Handling context"?

**Review by JrooTJunior — APPROVED** [review]

**Pull request merged**